src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2546,6 +2575,114 @@ export class StudyViewPageStore
+                },
+            ];
+
+            this.addMutationPlotAsChart(mutationPlotSpecs);
+            const chartAddedMessage = `Mutation Plot for ${hugoGeneSymbol} added.`;
+            this.launchToastMessage(chartAddedMessage);
+        } else {
+            this.changeChartVisibility(hugoGeneSymbol, false);
+            _.pull(this.visibleMutationPlotGenes, hugoGeneSymbol);
+
+            const chartRemovedMessage = `Mutation Plot for ${hugoGeneSymbol} removed.`;
+
+            this.launchToastMessage(chartRemovedMessage);
+        }
+    }
+
+    launchToastMessage(message: string): void {
+        toast.success(message, {
+            delay: 0,
+            position: 'top-right',
+            autoClose: 1500,
+            hideProgressBar: true,
+            closeOnClick: true,
+            pauseOnHover: true,
+            draggable: true,
+            progress: undefined,
+            theme: 'light',
+        });
+    }
+
+    public mutationPlotData = new MobxPromiseCache<
+        {
+            hugoGeneSymbol: string;
+        },
+        {
+            data: Mutation[];
+        }
+    >(q => ({
+        await: () => [this.selectedSamples, this.mutationProfiles],
+        invoke: async () => ({
+            data: await getMutationData(
+                this.selectedSamples.result,
+                this.mutationProfiles.result,
+                [q.hugoGeneSymbol]
+            ),
+        }),
+    }));
+
+    public createMutationStore(hugoGeneSymbol: string): void {
+        const data = this.mutationPlotData.get({ hugoGeneSymbol });
+        const store = new StudyViewMutationMapperStore(
+            { hugoGeneSymbol },
+            {},
+            () => data.result!.data
+        );
+
+        this.mutationPlotStore[hugoGeneSymbol] = store;
+    }
+
+    async updateStudyViewFilter(hugoGeneSymbol: string) {
+        const filteredMutationPlotData = this.mutationPlotStore[hugoGeneSymbol]
+            .samplesViaSelectedCodons;
+
+        if (filteredMutationPlotData.length) {
+            if (!this.filteredMutationPlots.has(hugoGeneSymbol)) {
+                const filteredSamples: SampleIdentifier[] = this.getMutationStore(
+                    hugoGeneSymbol
+                ).samplesViaSelectedCodons.map<SampleIdentifier>(
+                    (val: SampleData) => {
+                        return {
+                            studyId: val.studyId,
+                            sampleId: val.sampleId,
+                        };
+                    }
+                );
+
+                this.filteredMutationPlots.set(
+                    hugoGeneSymbol,
+                    this.getMutationStore(hugoGeneSymbol)
+                        .samplesViaSelectedCodons.length
+                );
+                this._chartSampleIdentifiersFilterSet.set(
+                    hugoGeneSymbol,
+                    filteredSamples
+                );
+            }
+        }
+    }
+
+    public getMutationStore(
```

Comment:
we should be careful when naming this function. this does not simply gets a mutation store. It inits a new one if it doesn't exits. so better rename it to make this clear.

```suggestion
    public getOrInitMutationStore(
```